### PR TITLE
Server architecture

### PR DESCRIPTION
**Pull request by mikrise2**

### CODE REVIEW

File: ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/database/models/data/ActivityData.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.jetbrains.research.tasktracker.database.models.data
+
+import org.jetbrains.exposed.sql.statements.InsertStatement
+
+object ActivityData : DataTable() {
```
Comment (nbirillo) [inline ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/database/models/data/ActivityData.kt:5]: I'd prefer to add descriptions to all db tables

File: ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/plugins/requests/UserRequests.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.jetbrains.research.tasktracker.database.models.User
+import org.jetbrains.research.tasktracker.database.models.Users
+
+fun Routing.createUser() {
+    post("/create-user") {
+        val formParameters = call.receiveParameters()
+        try {
+            val userId = transaction {
+                val name = formParameters.getOrFail("name")
```
Comment (nbirillo) [inline ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/plugins/requests/UserRequests.kt:21]: We need to return a user if db already contain a user with this email

Comment (nbirillo) [reply ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/plugins/requests/UserRequests.kt:21]: We should process it here or on the client side

File: ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/util/LogTableUtil.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+fun File.parseLogFile(logFileType: String, researchId: Int) {
+    CSVFormat.Builder.create().apply {
+        setHeader()
+        setSkipHeaderRecord(true)
+    }.build().parse(inputStream().bufferedReader())
+        .forEach {
+            transaction {
+                val table = when (logFileType) {
+                    "activity" -> ActivityData
```
Comment (nbirillo) [inline ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/util/LogTableUtil.kt:16]: I'd prefer to replace it with an enum class

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great!

File: ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/plugins/requests/UserRequests.kt (hunk 2 above)
Comment (mikrise2) [reply ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/plugins/requests/UserRequests.kt:21]: It is already resolved. Please check `getUserId` function below.

**Pull request merged**